crate/server/src/core/operations/export.rs

```diff
@@ -1,55 +1,89 @@
 use cosmian_kmip::kmip::{
     kmip_data_structures::{KeyMaterial, KeyValue},
     kmip_operations::{Export, ExportResponse},
-    kmip_types::StateEnumeration,
+    kmip_types::{KeyWrapType, StateEnumeration},
 };
-use cosmian_kms_utils::types::{ExtraDatabaseParams, ObjectOperationTypes};
+use cosmian_kms_utils::access::{ExtraDatabaseParams, ObjectOperationType};
 use tracing::trace;
 
 use crate::{
-    core::{operations::get::get_, KMS},
+    core::{
+        operations::{unwrap_key, wrapping::wrap_key},
+        KMS,
+    },
     error::KmsError,
     result::KResult,
 };
 
 /// Export an object
+///
 /// If the request contains a KeyWrappingData, the key will be wrapped
 /// If the request contains a KeyWrapType, the key will be unwrapped
 /// If both are present, the key will be wrapped
 /// If none are present, the key will be returned as is
-///
 pub async fn export(
     kms: &KMS,
     request: Export,
-    owner: &str,
+    user: &str,
     params: Option<&ExtraDatabaseParams>,
 ) -> KResult<ExportResponse> {
     trace!("Export: {}", serde_json::to_string(&request)?);
 
-    let uid = request
+    // there must be an identifier
+    let uid_or_tags = request
         .unique_identifier
-        .as_ref()
+        .clone()
         .ok_or(KmsError::UnsupportedPlaceholder)?;
 
-    let (mut object, state) = get_(
-        kms,
-        uid,
-        request.key_wrap_type,
-        request.key_wrapping_data,
-        owner,
-        params,
-        ObjectOperationTypes::Export,
-    )
-    .await?;
+    // retrieve from tags or use passed identifier
+    let mut owm_s = kms
+        .db
+        .retrieve(&uid_or_tags, user, ObjectOperationType::Export, params)
+        .await?;
+
+    // there can only be one object
+    let mut owm = match owm_s.len() {
+        0 => return Err(KmsError::ItemNotFound(uid_or_tags.to_owned())),
+        1 => owm_s.pop().expect("failed extracting the object"),
+        _ => {
+            return Err(KmsError::InvalidRequest(format!(
+                "too many objects for {uid_or_tags}",
+            )))
+        }
+    };
```

Comment:
This is at least the third time I see this piece of code. I already proposed an alternative form. Maybe you can also make it a dedicated procedure to avoid duplicates.